data/hooks/conf_regen/15-nginx

```diff
@@ -38,12 +38,18 @@ do_pre_regen() {
   for domain in $domain_list; do
     domain_conf_dir="${nginx_conf_dir}/${domain}.d"
     mkdir -p "$domain_conf_dir"
+    mkdir -p "/var/www/${domain}/.well-known/autoconfig/mail/"
```

Comment:
Thanks for this great piece of work !

I think it would be nice to avoid creating a folder `${domain}` in `/var/www/` for each domain ideally (sometimes, users have to browse it and it can become quite a mess if for some reason you have 10+ domains)

Instead, I propose that we use `/var/www/.well-known/${domain}/autoconfig/mail` ?

Also if we want this to be managed by the regenconf we should put this folder + files in `${pending_dir}`

so : `${pending_dir}/var/www/.well-known/${domain}/autoconfig/mail/`

What do you think ?

Reply:
I've updated the PR following your suggestion but I don't know how to make use of the magic of ${pending_dir}.

So currently it is broken, the following warning is displayed when adding a new domain: *Warning: The configuration file '/var/www/.well-known/domain.tld/autoconfig/mail/config-v1.1.xml' has been manually removed and will not be created*
The expected file tree is created in `/home/yunohost.conf/pending/nginx/var/www/.well-known/` but I don't know yet how to have it properly copied/moved into `/var/www/.well-known/`.

Can you please tell me how to use the ${pending_dir} correctly?

Note that I also added some cleanup code to *rm -rf* the `.well-known` dir upon domain deletion.
I don't think that it's related though.

Reply:
Indeed the regenconf is a bit non-trivial ;)

Basically, the regenconf scripts should always put files in $pending_dir and not in the true filesystem (at least during the "pre" step). Because it will then compare to the current file on the disk to check if it was manually edited. (And also because maybe the regen-conf is ran with --dry-run).

And each file you manage with the regen-conf, you should also delete with the regenconf. Otherwise, it will be considered as "manually removed".

To remove a file, you can see a few line below that there is a ` touch "${nginx_conf_dir}/${file}"`, i.e. creating an empty file, which corresponds to saying that it should be deleted. (That's a bit hackish, yup)

So in our case, I guess we want to do something similar, i.e. : 
- do a `ls /var/www/.well-known/*/autoconfig/mail/config-v1.1.xml'`
- for each file, extract the domain
- if the domain is not registered in yunohost anymore,
- put an empty file at the same place in $pending_dir

If we want to go further, we could want to completely remove /var/www/.well-known/$domain ... but I don't know if we have a mechanism to remove entire directories... For instance I suspect that, after you remove a domain, there is still a folder /etc/nginx/conf.d/domain.tld.d/ ... maybe that's no bigdeal.